File: pag_darwin/darwin/Classes/PAGCompositionImpl.swift
//
//  PAGCompositionImpl.swift
//
//  Created by 闫守旺 on 2025/4/9.
//

import libpag

class PAGCompositionImpl: PigeonApiDelegatePAGCompositionApi {
    func getWidth(pigeonApi: PigeonApiPAGCompositionApi, pigeonInstance: PAGComposition) throws -> Int64 {
        return pigeonInstance.width().api
    }
    
    func getHeight(pigeonApi: PigeonApiPAGCompositionApi, pigeonInstance: PAGComposition) throws -> Int64 {
        return pigeonInstance.height().api
    }
}

File: pag_android/android/src/main/kotlin/dev/hebei/pag_android/PAGCompositionImpl.kt
package dev.hebei.pag_android

import org.libpag.PAGComposition

class PAGCompositionImpl(registrar: PAGApiPigeonProxyApiRegistrar) : PigeonApiPAGCompositionApi(registrar) {
    override fun getWidth(pigeon_instance: PAGComposition): Long {
        return pigeon_instance.width().api
    }

    override fun getHeight(pigeon_instance: PAGComposition): Long {
        return pigeon_instance.height().api
    }
}